From: Hugo Pereira Da Costa <hugo.pereira@free.fr>
Date: Sun, 09 Mar 2014 12:00:33 +0000
Subject: make sure parent is registered to innerShadowEngine before registering child
X-Git-Url: http://quickgit.kde.org/?p=oxygen-gtk.git&a=commitdiff&h=b39931fa3ad7a2756bcb1494d5c3559513406ce2
---
make sure parent is registered to innerShadowEngine before registering child
CCBUG: 331879
---


--- a/src/oxygenstylewrapper.cpp
+++ b/src/oxygenstylewrapper.cpp
@@ -1669,7 +1669,10 @@
             }
 
             // register to inner shadow engine
-            if( shadow == GTK_SHADOW_IN && gtk_scrolled_window_get_shadow_type( GTK_SCROLLED_WINDOW( widget ) ) == GTK_SHADOW_IN )
+            if(
+                shadow == GTK_SHADOW_IN &&
+                gtk_scrolled_window_get_shadow_type( GTK_SCROLLED_WINDOW( widget ) ) == GTK_SHADOW_IN &&
+                Style::instance().animations().innerShadowEngine().contains( widget ) )
             { Style::instance().animations().innerShadowEngine().registerChild( widget, gtk_bin_get_child( GTK_BIN( widget ) ) ); }
 
         } else if(
@@ -2074,7 +2077,8 @@
                     if( gtk_scrolled_window_get_shadow_type( scrolledWindow ) != GTK_SHADOW_IN )
                     {
                         gtk_scrolled_window_set_shadow_type( scrolledWindow, GTK_SHADOW_IN );
-                        Style::instance().animations().innerShadowEngine().registerChild( child, gtk_bin_get_child( GTK_BIN( child ) ) );
+                        if( Style::instance().animations().innerShadowEngine().contains( child ) )
+                        { Style::instance().animations().innerShadowEngine().registerChild( child, gtk_bin_get_child( GTK_BIN( child ) ) ); }
                     }
 
                     return;

